Conference-Android/app/src/main/java/com/sagetech/conference_android/app/ui/activities/ConferenceSessionListActivity.java

```diff
@@ -0,0 +1,168 @@
+
+                        Intent eventDetailIntent = new Intent(view.getContext(), EventDetailActivity.class);
+                        eventDetailIntent.putExtra("id", (Long)view.getTag());
+                        startActivity(eventDetailIntent);
+                    }
+                })
+        );
+
+        conferenceId = getIntent().getExtras().getInt("id");
+
+        presenter = new ConferenceSessionListActivityPresenter(this, conferenceController, Long.valueOf(conferenceId));
+        presenter.initialize();
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            // Respond to the action bar's Up/Home button
+            case android.R.id.home:
+                NavUtils.navigateUpFromSameTask(this);
+                return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        presenter.onDestroy();
+    }
+
+    public class ConferenceSessionsAdapter extends RecyclerView.Adapter<ConferenceSessionsAdapter.ViewHolder> {
+        private final List<ConferenceSessionData> conferenceSessionDatas;
+
+        private final String TIME_FORMAT = "h:mm a";
+        private final String DAY_FORMAT = "EEEE, MMMM dd, yyyy";
+
+        public class ViewHolder extends RecyclerView.ViewHolder {
+            // each data item is just a string in this case
+            public ViewHolder(View v) {
+                super(v);
+            }
+        }
+
+        public ConferenceSessionsAdapter(List<ConferenceSessionData> conferenceSessionDatas) {
+            this.conferenceSessionDatas = conferenceSessionDatas;
+        }
+
+        @Override
+        public ConferenceSessionsAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+            View v = LayoutInflater.from(parent.getContext())
+                    .inflate(R.layout.list_view_item, parent, false);
+
+            ViewHolder vh = new ViewHolder(v);
+            return vh;
+        }
+
+        @Override
+        public void onBindViewHolder(ConferenceSessionsAdapter.ViewHolder holder, int position) {
+            holder.itemView.setTag(conferenceSessionDatas.get(position).getId());
+
+            TextView day = getTextViewById(holder, R.id.day);
+            day.setText(toDay(conferenceSessionDatas.get(position).getStartDttm()));
+
+            TextView time = getTextViewById(holder, R.id.time);
+            time.setText(toTime(conferenceSessionDatas.get(position).getStartDttm()));
+
+            TextView title = getTextViewById(holder, R.id.title);
+            title.setText(conferenceSessionDatas.get(position).getName());
+
+            TextView room = getTextViewById(holder, R.id.room);
+            room.setText("112E"); //TODO -- set to real value once roomdata is available
+        }
+
+        private TextView getTextViewById(ConferenceSessionsAdapter.ViewHolder holder, int id) {
```

Comment:
Additionally, I would probably abstract these away a little bit more, so your ViewHolder would have methods to `setTitle()`, `setStartDateTime()`, and `setRoom()`, and move the logic for formatting that into the ViewHolder itself, instead of the adapter.